packages/opentelemetry-core/src/utils/environment.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { LogLevel } from '../common/types';
+
+export type ENVIRONMENT_MAP = { [key: string]: string | number };
+
+/**
+ * Environment interface to define all names
+ */
+export interface ENVIRONMENT {
+  OTEL_LOG_LEVEL?: LogLevel;
+  OTEL_NO_PATCH_MODULES?: string;
+  OTEL_SAMPLING_PROBABILITY?: number;
+}
+
+const ENVIRONMENT_NUMBERS: Partial<keyof ENVIRONMENT>[] = [
+  'OTEL_LOG_LEVEL',
+  'OTEL_SAMPLING_PROBABILITY',
+];
+
+/**
+ * Default environment variables
+ */
+export const DEFAULT_ENVIRONMENT: ENVIRONMENT = {
+  OTEL_NO_PATCH_MODULES: '',
+  OTEL_LOG_LEVEL: LogLevel.ERROR,
+  OTEL_SAMPLING_PROBABILITY: 1,
+};
+
+/**
+ * Parses a variable as number with number validation
+ * @param name
+ * @param environment
+ * @param values
+ * @param min
+ * @param max
+ */
+function parseNumber(
```

Comment:
what is this difference between this and `parseInt`?

Reply:
hmm it does much more, have you seen the body of function ?

Reply:
yes. actually I think using `parseFloat(values[name])` might be better than `Number(values[name])` here so that you get `NaN` on falsey values.

Reply:
pls see the earlier thread about the same issue